src/internal/proc.js

```diff
@@ -178,9 +192,10 @@ export default function proc(
     Creates a new task descriptor for this generator, We'll also create a main task
     to track the main flow (besides other forked tasks)
   **/
-  const task = newTask(parentEffectId, name, iterator, cont)
-  const mainTask = { name, cancel: cancelMain, isRunning: true }
-  const taskQueue = forkQueue(name, mainTask, end)
+  const task = newTask(parentEffectId, meta, iterator, cont)
+  // NOTE: not sure where tasks are used, so left the mainTask interface as is
```

Comment:
what do you mean by "where tasks are used"? how did it matter for the implementation?

Reply:
On the one hand, we don't use `name` as independent property, but part of `meta` (code consistency). On the other hand, I'm not sure whether we need to create `meta` descriptor for `mainTask` (what's location in that case?)

Reply:
`mainTask` is completely internal, see [this comment](https://github.com/redux-saga/redux-saga/pull/1263#issuecomment-347485969). I dont think it needs to have any meta attached to it (regarding location etc)